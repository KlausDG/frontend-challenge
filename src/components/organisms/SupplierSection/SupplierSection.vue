<script setup lang="ts">
import { Badge, Card, Stack, Typography } from '@/components/atoms'
import { IconText } from '@/components/molecules'
import type { Supplier } from '@/models'

defineProps<{
  data: Supplier
}>()
</script>

<template>
  <section class="supplier__container">
    <Card title="Supplier">
      <template #subtitle>
        <Stack :spacing="8" class="subtitle__container">
          <Typography tag="p" variant="semibold" color="dark">{{ data.name }}</Typography>
          <Badge>#{{ data.code }}</Badge>
        </Stack>
      </template>
      <Stack class="info__container">
        <Stack direction="column" :spacing="4" class="content__column">
          <IconText
            icon="file-landscape-alt"
            :text="`${data.document.type}: ${data.document.value}`"
          />
          <IconText icon="map-marker" :text="data.address" />
          <IconText icon="user" :text="data.contact.name" />
        </Stack>
        <Stack direction="column" :spacing="4">
          <IconText icon="envelope" :text="data.contact.email" />
          <IconText icon="phone" :text="data.contact.phone" />
          <IconText icon="print" :text="data.contact.fax" />
          <IconText icon="eye" :text="data.readAt" />
        </Stack>
      </Stack>
    </Card>
  </section>
</template>

<style lang="scss" scoped>
.supplier__container {
  @media (max-width: 1024px) {
    padding-left: 12px;
    padding-right: 12px;
  }
}

.subtitle__container {
  align-items: center;
}

.info__container {
  @media (max-width: 1024px) {
    flex-direction: column !important;
    gap: 8px !important;
  }
}

.content__column {
  width: 50%;

  @media (max-width: 1024px) {
    width: 100%;
  }
}
</style>
