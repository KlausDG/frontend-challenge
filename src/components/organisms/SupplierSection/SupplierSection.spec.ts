import { mount } from '@vue/test-utils'
import { describe, it, expect } from 'vitest'
import SupplierSection from './SupplierSection.vue'

const CardMock = {
  name: 'Card',
  props: ['title'],
  template: '<div class="mock-card"><slot name="subtitle" /><slot /></div>',
}

const StackMock = {
  name: 'Stack',
  props: ['direction', 'spacing', 'class'],
  template: '<div class="mock-stack"><slot /></div>',
}

const TypographyMock = {
  name: 'Typography',
  props: ['tag', 'variant', 'color'],
  template: '<div class="mock-typography"><slot /></div>',
}

const BadgeMock = {
  name: 'Badge',
  props: [''],
  template: '<div class="mock-badge"><slot /></div>',
}

const IconTextMock = {
  name: 'IconText',
  props: ['icon', 'text'],
  template: '<div class="mock-icon-text">{{ icon }}:{{ text }}</div>',
}

describe('SupplierSection.vue', () => {
  const supplierData = {
    name: 'Supplier Inc',
    code: 'SUP001',
    address: '123 Supplier Rd',
    contact: {
      name: 'Alice Johnson',
      email: 'alice@example.com',
      phone: '111222333',
      fax: '999888777',
    },
    document: {
      type: 'CNPJ',
      value: '12.345.678/0001-90',
    },
    readAt: '2025-08-27',
  }

  const wrapper = mount(SupplierSection, {
    props: { data: supplierData },
    global: {
      components: {
        Card: CardMock,
        Stack: StackMock,
        Typography: TypographyMock,
        Badge: BadgeMock,
        IconText: IconTextMock,
      },
    },
  })

  it('renders Card with title "Supplier"', () => {
    const card = wrapper.findComponent(CardMock)
    expect(card.exists()).toBe(true)
    expect(card.props('title')).toBe('Supplier')
  })

  it('renders subtitle with Typography and Badge', () => {
    const typography = wrapper.findComponent(TypographyMock)
    expect(typography.exists()).toBe(true)

    const badge = wrapper.findComponent(BadgeMock)
    expect(badge.exists()).toBe(true)
    expect(badge.text()).toBe(`#${supplierData.code}`)
  })

  it('renders all IconText components with correct props', () => {
    const iconTexts = wrapper.findAllComponents(IconTextMock)
    expect(iconTexts).toHaveLength(7)

    expect(iconTexts[0].props()).toEqual({
      icon: 'file-landscape-alt',
      text: `${supplierData.document.type}: ${supplierData.document.value}`,
    })
    expect(iconTexts[1].props()).toEqual({ icon: 'map-marker', text: supplierData.address })
    expect(iconTexts[2].props()).toEqual({ icon: 'user', text: supplierData.contact.name })
    expect(iconTexts[3].props()).toEqual({ icon: 'envelope', text: supplierData.contact.email })
    expect(iconTexts[4].props()).toEqual({ icon: 'phone', text: supplierData.contact.phone })
    expect(iconTexts[5].props()).toEqual({ icon: 'print', text: supplierData.contact.fax })
    expect(iconTexts[6].props()).toEqual({ icon: 'eye', text: supplierData.readAt })
  })

  it('renders supplier__container, info__container and content__column classes', () => {
    expect(wrapper.find('.supplier__container').exists()).toBe(true)
    expect(wrapper.find('.info__container').exists()).toBe(true)
    expect(wrapper.find('.content__column').exists()).toBe(true)
  })

  it('matches the component snapshot', () => {
    expect(wrapper.html()).toMatchSnapshot()
  })
})
