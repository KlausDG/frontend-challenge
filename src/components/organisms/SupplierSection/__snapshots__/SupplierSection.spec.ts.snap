// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`SupplierSection.vue > matches the component snapshot 1`] = `
"<section data-v-65d85fd2="" class="supplier__container">
  <div data-v-69f42c1a="" data-v-9bb7b3b9="" data-v-65d85fd2="" class="stack__container card__container" style="flex-direction: column; gap: 16px;">
    <div data-v-69f42c1a="" data-v-9bb7b3b9="" class="stack__container" style="flex-direction: column; gap: 4px;">
      <p data-v-9bb7b3b9="" class="typography _caption_d0cbd2 _text_d0cbd2">Supplier</p>
      <div data-v-69f42c1a="" data-v-65d85fd2="" class="stack__container subtitle__container" style="flex-direction: row; gap: 8px;">
        <p data-v-65d85fd2="" class="typography _semibold_d0cbd2 _dark_d0cbd2">Supplier Inc</p>
        <div data-v-df2d503b="" data-v-65d85fd2="" class="badge__container">
          <p data-v-df2d503b="" class="typography _caption_d0cbd2 _primary_d0cbd2">#SUP001</p>
        </div>
      </div>
    </div>
    <div data-v-69f42c1a="" data-v-65d85fd2="" class="stack__container info__container" style="flex-direction: row; gap: 0px;">
      <div data-v-69f42c1a="" data-v-65d85fd2="" class="stack__container content__column" style="flex-direction: column; gap: 4px;">
        <div data-v-70c01fa4="" data-v-65d85fd2="" class="icon-text__container">
          <unicon data-v-70c01fa4="" name="file-landscape-alt"></unicon>
          <p data-v-70c01fa4="" class="typography _caption_d0cbd2 _text_d0cbd2">CNPJ: 12.345.678/0001-90</p>
        </div>
        <div data-v-70c01fa4="" data-v-65d85fd2="" class="icon-text__container">
          <unicon data-v-70c01fa4="" name="map-marker"></unicon>
          <p data-v-70c01fa4="" class="typography _caption_d0cbd2 _text_d0cbd2">123 Supplier Rd</p>
        </div>
        <div data-v-70c01fa4="" data-v-65d85fd2="" class="icon-text__container">
          <unicon data-v-70c01fa4="" name="user"></unicon>
          <p data-v-70c01fa4="" class="typography _caption_d0cbd2 _text_d0cbd2">Alice Johnson</p>
        </div>
      </div>
      <div data-v-69f42c1a="" data-v-65d85fd2="" class="stack__container" style="flex-direction: column; gap: 4px;">
        <div data-v-70c01fa4="" data-v-65d85fd2="" class="icon-text__container">
          <unicon data-v-70c01fa4="" name="envelope"></unicon>
          <p data-v-70c01fa4="" class="typography _caption_d0cbd2 _text_d0cbd2">alice@example.com</p>
        </div>
        <div data-v-70c01fa4="" data-v-65d85fd2="" class="icon-text__container">
          <unicon data-v-70c01fa4="" name="phone"></unicon>
          <p data-v-70c01fa4="" class="typography _caption_d0cbd2 _text_d0cbd2">111222333</p>
        </div>
        <div data-v-70c01fa4="" data-v-65d85fd2="" class="icon-text__container">
          <unicon data-v-70c01fa4="" name="print"></unicon>
          <p data-v-70c01fa4="" class="typography _caption_d0cbd2 _text_d0cbd2">999888777</p>
        </div>
        <div data-v-70c01fa4="" data-v-65d85fd2="" class="icon-text__container">
          <unicon data-v-70c01fa4="" name="eye"></unicon>
          <p data-v-70c01fa4="" class="typography _caption_d0cbd2 _text_d0cbd2">2025-08-27</p>
        </div>
      </div>
    </div>
  </div>
</section>"
`;
